tests/python/unittest/test_subgraph_op.py

```diff
@@ -18,351 +18,448 @@
+    
+def check_subgraph(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe1(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe2(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe3(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe4(sym[0], subgraph_backend, op_names)
+
+def check_subgraph_backend_sym(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe5(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe6(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe7(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe8(sym[0], subgraph_backend, op_names)
+
+def check_subgraph_backend_gluon(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe9(sym, subgraph_backend, op_names)
+
+def test_subgraph():
+    check_subgraph('default')
+
+def test_subgraph_v2():
+    check_subgraph('default_v2')
+
+def test_subgraph_backend_sym():
+    check_subgraph_backend_sym('default')
+
+def test_subgraph_backend_sym_v2():
+    check_subgraph_backend_sym('default_v2')
+
+def test_subgraph_backend_gluon():
+    check_subgraph_backend_gluon('default')
+
+def test_subgraph_backend_gluon_v2():
+    check_subgraph_backend_gluon('default_v2')
+
+def test_subgraph_backend_gluon_ext1():
+    def get_net():
+        net = nn.HybridSequential()  # Here we use the class HybridSequential.
+        net.add(nn.Dense(256, activation='relu'),
+                nn.Dense(128, activation='relu'),
+                nn.Dense(2))
+        return net
+
+    x = nd.random.normal(shape=(1, 512))
+    net = get_net()
+    net.collect_params().initialize()
+    outputs1 = net(x)
+
+    subgraph_backend = 'default'
+    op_names = []
+    check_call(_LIB.MXSetSubgraphPropertyOpNamesV2(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                c_str_array(op_names)))
+    net.hybridize(backend = subgraph_backend)
+    outputs2 = net(x)
+    check_call(_LIB.MXRemoveSubgraphPropertyOpNamesV2(c_str(subgraph_backend)))
+
+    # compare outputs
+    assert len(outputs1) == len(outputs2)
+    for i in range(len(outputs1)):
+        assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+
+def test_subgraph_backend_gluon_ext2():
+    class Net(gluon.HybridBlock):
+        def __init__(self, **kwargs):
+            super(Net, self).__init__(**kwargs)
+            with self.name_scope():
+                self.fc1 = nn.Dense(256)
+                self.fc2 = nn.Dense(128)
+                self.fc3 = nn.Dense(2)
+
+        def hybrid_forward(self, F, x):
+            x = F.relu(self.fc1(x))
+            x = F.relu(self.fc2(x))
+            return self.fc3(x)
+
+    x = nd.random.normal(shape=(1, 512))
+    net = Net()
+    net.collect_params().initialize()
+    outputs1 = net(x)
```

Comment:
same deal here, you're running an inference before partitioning

Reply:
Wiil add tests without the inference before partition and refactor tests.

Reply:
i changed it so the block is recreated before we partition and weights are copied over, so results can be compared